base/dcp_receiver.go

```diff
@@ -284,11 +286,14 @@ func StartDCPFeed(bucket Bucket, spec BucketSpec, args sgbucket.FeedArguments, c
 		auth = NoPasswordAuthHandler{handler: spec.Auth}
 	}
 
-	// If using TLS, pass a custom connect method to support using TLS for cbdatasource's memcached connections
-	if spec.IsTLS() {
-		dataSourceOptions.Connect = spec.TLSConnect
+	dataSourceOptions.TLSConfig = func() *tls.Config {
+		return spec.TLSConfig()
 	}
 
+	// A lookup of host dest to external alternate address hostnames
+	externalAlternateAddresses := make(map[string]string)
```

Comment:
I find the scoping for externalAlternateAddresses a bit difficult to reason about, but I'm assuming the reason your instantiating here (as opposed to in alternateAddressShims) is required for scoping?  If so, maybe stick a comment here with some details?

Reply:
I think this is an oversight from when the code used to live here actually. I'll retain the existing make inside the shim helper func.